shared/chat/payments/status/index.js

```diff
@@ -24,18 +37,78 @@ const getIcon = status => {
   }
 }
 
-const PaymentStatus = (props: Props) => (
-  <Kb.Text type="BodyExtrabold" allowFontScaling={!!props.allowFontScaling} style={styles[props.status]}>
-    {' '}
-    {props.text}{' '}
-    <Kb.Icon
-      type={getIcon(props.status)}
-      fontSize={12}
-      boxStyle={styles.iconBoxStyle}
-      style={styles[props.status + 'Icon']}
-    />{' '}
-  </Kb.Text>
-)
+class PaymentStatus extends React.Component<Props, State> {
+  statusRef: any
+  state = {showPopup: false}
+  constructor(props: Props) {
+    super(props)
+    this.statusRef = React.createRef()
+  }
+  _showPopup = () => {
+    if (this.props.allowPopup) {
+      this.setState({showPopup: true})
+    }
+  }
+  _hidePopup = () => {
+    this.setState({showPopup: false})
+  }
+  _getAttachmentRef = () => {
+    return this.statusRef.current
+  }
+  render() {
+    const text = (
+      <Kb.Text
+        ref={this.statusRef}
+        type="BodyExtrabold"
+        allowFontScaling={!!this.props.allowFontScaling}
+        style={styles[this.props.status]}
+        onClick={this._showPopup}
+      >
+        {' '}
+        {this.props.text}{' '}
+        <Kb.Icon
+          type={getIcon(this.props.status)}
+          fontSize={12}
+          boxStyle={styles.iconBoxStyle}
+          style={styles[this.props.status + 'Icon']}
```

Comment:
fyi this likely doesn't flow correctly. it can't dynamically understand the type. this is fine short term but a better approach is to do like 
``` 
this.getStatusIconStyle = () => {
 switch(this.props.status) {
case 'whatever':
return 'whateverIcon'
}
}
```

so flow can know all the cases of the map